File: oracle/ruby/update/oracle_update.rb
#! /usr/bin/ruby
#
#	oracle_update.rb
#
#				May/31/2011
#
require 'rubygems'
require 'oci8'
require 'date'
#
# ------------------------------------------------------------
def	disp_proc (dbi)
#
cur = dbi.exec('select * from cities')
while   row = cur.fetch
	puts row.join("\t")
end
end
#
# ------------------------------------------------------------
def	update_proc (dbi,id,population)
date_mod=Date.today
#date_mod=Time.now
#puts date_mod
sql_str="UPDATE cities SET population='#{population}', DATE_MOD='#{date_mod}' where ID = '#{id}'"
dbi.exec(sql_str)
end
# ------------------------------------------------------------
puts	"*** 開始 ***"
id_in = ARGV[0]
population_in = ARGV[1].to_i
#
puts id_in,population_in
#
dbi=OCI8.new('scott', 'tiger', '//spn109:1521/xe')
#
update_proc(dbi,id_in,population_in)
dbi.commit()
#
disp_proc (dbi)
#
puts	"*** 終了 ***"
#
# ------------------------------------------------------------

File: oracle/clojure/update/oracle_update.clj
; -----------------------------------------------------------------
;
;	oracle_update.clj
;
;					Sep/29/2010
;
; -----------------------------------------------------------------
(import '(java.util Date))
(import '(java.text SimpleDateFormat))
(use 'clojure.contrib.sql)
(use '[clojure.contrib.str-utils :only (str-join)])
(println "*** 開始 ***")

(let [db-host "spn109:1521"
	db-name "xe"
	id (first *command-line-args*)
	population (second *command-line-args*)
	today (. (new SimpleDateFormat "yyyy-MM-dd") format (new Date))
	]
(println id)
(println population)
(println today)

  (def db {:classname "oracle.jdbc.driver.OracleDriver"
           :subprotocol "oracle:thin"
           :subname (str ":@" db-host "/" db-name)
           :user "scott"
           :password "tiger"})

  (with-connection db
    (update-values :cities ["id=?" id]
	{:population population :date_mod today})
      ))
(println "*** 終了 ***")
; -----------------------------------------------------------------
